lib/datapacksjob.js

```diff
@@ -780,8 +785,9 @@ DataPacksJob.prototype.buildManifestFromQueries = async function(jobInfo) {
                 if (!queryData || (!jobInfo.specificManifestKeys && queryData.manifestOnly)) {
                     continue;
                 }
-
-                allQueries.push( { context: this, func: 'runQueryForManifest', argument: { jobInfo: jobInfo, queryData: queryData } });
+                if((jobInfo.versionCompare && queryData.versionCompare) || (!jobInfo.versionCompare && !queryData.versionCompare)){
```

Comment:
Should just change this to return early `if (queryData.versionCompare && !jobInfo.versionCompare)`